lib/WorseReferenceFinder/WorseReflectionDefinitionLocator.php

```diff
@@ -245,4 +252,42 @@ private function gotoMember(NodeContext $symbolContext): TypeLocations
 
         return new TypeLocations($locations);
     }
+
+    private function gotoMethodDeclaration(MemberDeclarationContext $symbolContext): TypeLocations
+    {
+        try {
+            $class = $this->reflector->reflectClass($symbolContext->classType()->name());
+        } catch (NotFound $notFound) {
+            return new TypeLocations([]);
+        }
+
+        // find first parent definition or return declaring class
+        $member = (function (string $name) use ($class) {
+            foreach ((new ClassHierarchyResolver())->resolve($class) as $currentClass) {
+                if ($currentClass->ownMembers()->has($name)) {
+                    return $currentClass->ownMembers()->byName($name)->first();
+                }
+            }
+            return null;
+        })($symbolContext->name());
+
+        if (null === $member) {
+            return new TypeLocations([]);
+        }
+
+
+        $path = $member->declaringClass()->sourceCode()->path();
+
+        if (null === $path) {
+            throw new CouldNotLocateDefinition(sprintf(
+                'The source code for class "%s" has no path associated with it.',
+                (string) $member->declaringClass()->name()
+            ));
+        }
+
+        return new TypeLocations([new TypeLocation($symbolContext->classType(), new Location(
```

Comment:
we culd quite easily return all the places where the method was "defined" here (e.g. abstract class, interface, trait...)